### ISSUE TITLE
Add field validation for obvious fields 

### ISSUE THREAD

**Issue by DiamondJoseph** [DiamondLightSource/i22-bluesky#80]
e.g. numbers > 0, detectors not empty

**Comment by stan-dot** [comment]
do you see this implemented with pydantic @DiamondJoseph? and if yes, why here and not in dodal?

**Comment by DiamondJoseph** [comment]
Yes with Pydantic Field types, and here because here is where plans are written. We can't validate the i22 specific plans that are written in this module from dodal: we can validate them with the types defined in dodal.
